File: lib/cfr/builtins/libs/flutter/painting/edgeInsetsGeometry.dart
import 'dart:core' as _fac9;
import 'dart:ui' as _a643;

import 'package:flutter/src/painting/basic_types.dart' as _487f;
import 'package:flutter/src/painting/edge_insets.dart' as _96d1;

import 'package:hydro_sdk/cfr/runtimeSupport.dart' as _36c2;

class VMManagedEdgeInsetsGeometry
    extends _36c2.VMManagedBox<_96d1.EdgeInsetsGeometry> {
  VMManagedEdgeInsetsGeometry(
      {required this.table, required this.vmObject, required this.hydroState})
      : super(
          table: table,
          vmObject: vmObject,
          hydroState: hydroState,
        ) {
    table['getIsNonNegative'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.isNonNegative,
      ];
    });
    table['getHorizontal'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.horizontal,
      ];
    });
    table['getVertical'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.vertical,
      ];
    });
    table['along'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.along(_36c2.maybeUnBoxEnum(
            values: _487f.Axis.values, boxedEnum: luaCallerArguments[1])),
      ];
    });
    table['getCollapsedSize'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_a643.Size>(
            object: vmObject.collapsedSize,
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['getFlipped'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(
            object: vmObject.flipped,
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['inflateSize'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_a643.Size>(
            object: vmObject.inflateSize(
                _36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(
                    luaCallerArguments[1],
                    parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['deflateSize'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_a643.Size>(
            object: vmObject.deflateSize(
                _36c2.maybeUnBoxAndBuildArgument<_a643.Size, _fac9.dynamic>(
                    luaCallerArguments[1],
                    parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['subtract'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(
            object: vmObject.subtract(_36c2.maybeUnBoxAndBuildArgument<
                _96d1.EdgeInsetsGeometry,
                _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['add'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(
            object: vmObject.add(_36c2.maybeUnBoxAndBuildArgument<
                _96d1.EdgeInsetsGeometry,
                _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['clamp'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry>(
            object: vmObject.clamp(
                _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry,
                        _fac9.dynamic>(luaCallerArguments[1],
                    parentState: hydroState),
                _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry,
                        _fac9.dynamic>(luaCallerArguments[2],
                    parentState: hydroState)),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['resolve'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        _36c2.maybeBoxObject<_96d1.EdgeInsets>(
            object: vmObject.resolve(_36c2.maybeUnBoxEnum(
                values: _a643.TextDirection.values,
                boxedEnum: luaCallerArguments[1])),
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    });
    table['toString'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.toString(),
      ];
    });
    table['getHashCode'] = _36c2.makeLuaDartFunc(
        func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
      return [
        vmObject.hashCode,
      ];
    });
  }

  final _36c2.HydroTable table;

  final _36c2.HydroState hydroState;

  final _96d1.EdgeInsetsGeometry vmObject;
}

void loadEdgeInsetsGeometry(
    {required _36c2.HydroState hydroState, required _36c2.HydroTable table}) {
  table['edgeInsetsGeometryLerp'] = _36c2.makeLuaDartFunc(
      func: (_fac9.List<_fac9.dynamic> luaCallerArguments) {
    final returnValue = _96d1.EdgeInsetsGeometry.lerp(
        _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry?,
            _fac9.dynamic>(luaCallerArguments[1], parentState: hydroState),
        _36c2.maybeUnBoxAndBuildArgument<_96d1.EdgeInsetsGeometry?,
            _fac9.dynamic>(luaCallerArguments[2], parentState: hydroState),
        luaCallerArguments[3]?.toDouble());
    if (returnValue != null) {
      return [
        _36c2.maybeBoxObject<_96d1.EdgeInsetsGeometry?>(
            object: returnValue,
            hydroState: hydroState,
            table: _36c2.HydroTable()),
      ];
    }
    return [];
  });
  _36c2.registerBoxer<_96d1.EdgeInsetsGeometry>(boxer: (
      {required _96d1.EdgeInsetsGeometry vmObject,
      required _36c2.HydroState hydroState,
      required _36c2.HydroTable table}) {
    return VMManagedEdgeInsetsGeometry(
        vmObject: vmObject, hydroState: hydroState, table: table);
  });
}

File: runtime/flutter/painting/edgeInsetsGeometry.ts
import { ISize } from "../../dart/ui/size";
import { TextDirection } from "../../dart/ui/textDirection";
import { Axis } from "./axis";
import { IEdgeInsets } from "./edgeInsets";
declare const flutter: {
    painting: {
        edgeInsetsGeometryLerp: (
            a: IEdgeInsetsGeometry | undefined,
            b: IEdgeInsetsGeometry | undefined,
            t: number
        ) => IEdgeInsetsGeometry | undefined;
    };
};
export interface IEdgeInsetsGeometry {
    getIsNonNegative: () => boolean;
    getHorizontal: () => number;
    getVertical: () => number;
    along: (axis: Axis) => number;
    getCollapsedSize: () => ISize;
    getFlipped: () => IEdgeInsetsGeometry;
    inflateSize: (size: ISize) => ISize;
    deflateSize: (size: ISize) => ISize;
    subtract: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
    add: (other: IEdgeInsetsGeometry) => IEdgeInsetsGeometry;
    clamp: (
        min: IEdgeInsetsGeometry,
        max: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry;
    resolve: (direction?: TextDirection | undefined) => IEdgeInsets;
    toString: () => string;
    getHashCode: () => number;
}
export class EdgeInsetsGeometry {
    public static lerp(
        a: IEdgeInsetsGeometry | undefined,
        b: IEdgeInsetsGeometry | undefined,
        t: number
    ): IEdgeInsetsGeometry | undefined {
        return flutter.painting.edgeInsetsGeometryLerp(a, b, t);
    }
    private readonly _dart_getIsNonNegative: () => boolean = undefined as any;
    private readonly _dart_getHorizontal: () => number = undefined as any;
    private readonly _dart_getVertical: () => number = undefined as any;
    private readonly _dart_along: (axis: Axis) => number = undefined as any;
    private readonly _dart_getCollapsedSize: () => ISize = undefined as any;
    private readonly _dart_getFlipped: () => IEdgeInsetsGeometry =
        undefined as any;
    private readonly _dart_inflateSize: (size: ISize) => ISize =
        undefined as any;
    private readonly _dart_deflateSize: (size: ISize) => ISize =
        undefined as any;
    private readonly _dart_subtract: (
        other: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_add: (
        other: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_clamp: (
        min: IEdgeInsetsGeometry,
        max: IEdgeInsetsGeometry
    ) => IEdgeInsetsGeometry = undefined as any;
    private readonly _dart_resolve: (
        direction?: TextDirection | undefined
    ) => IEdgeInsets = undefined as any;
    private readonly _dart_toString: () => string = undefined as any;
    private readonly _dart_getHashCode: () => number = undefined as any;
    public getIsNonNegative(): boolean {
        return this._dart_getIsNonNegative();
    }
    public getHorizontal(): number {
        return this._dart_getHorizontal();
    }
    public getVertical(): number {
        return this._dart_getVertical();
    }
    public along(axis: Axis): number {
        return this._dart_along(axis);
    }
    public getCollapsedSize(): ISize {
        return this._dart_getCollapsedSize();
    }
    public getFlipped(): IEdgeInsetsGeometry {
        return this._dart_getFlipped();
    }
    public inflateSize(size: ISize): ISize {
        return this._dart_inflateSize(size);
    }
    public deflateSize(size: ISize): ISize {
        return this._dart_deflateSize(size);
    }
    public subtract(other: IEdgeInsetsGeometry): IEdgeInsetsGeometry {
        return this._dart_subtract(other);
    }
    public add(other: IEdgeInsetsGeometry): IEdgeInsetsGeometry {
        return this._dart_add(other);
    }
    public clamp(
        min: IEdgeInsetsGeometry,
        max: IEdgeInsetsGeometry
    ): IEdgeInsetsGeometry {
        return this._dart_clamp(min, max);
    }
    public resolve(direction?: TextDirection | undefined): IEdgeInsets {
        return this._dart_resolve(direction);
    }
    public toString(): string {
        return this._dart_toString();
    }
    public getHashCode(): number {
        return this._dart_getHashCode();
    }
}
